### PR TITLE
Second element of Time::HiRes::gettimeofday() is expected to be an integer

### PR DESCRIPTION
**Pull request by astj**
With current implementation (that I implemented in #2 ), the second element of `gettimeofday` sometimes becomes a decimal, but that of "real" `Time::HiRes::gettimeofday()` is integer. (refs: https://metacpan.org/source/RJBS/Time-HiRes-1.9728/HiRes.xs#L208-209 )

### CODE REVIEW

**Comment by astj** [comment]
@tarao Please review this p-r :bow:

File: lib/Test/MockTime/HiRes.pm [hunk 1]
```diff
@@ -80,7 +80,7 @@ sub gettimeofday() {
     if (defined $Test::MockTime::fixed) {
         return wantarray ? do {
             my $int_part = int($Test::MockTime::fixed);
-            ($int_part, 1_000_000 * ($Test::MockTime::fixed - $int_part))
+            ($int_part, 1_000_000 * sprintf('%.6f', ($Test::MockTime::fixed - $int_part)))
```
Comment (tarao) [inline lib/Test/MockTime/HiRes.pm:82]: Why is this not `int(1_000_000 * ($Test::MockTime::fixed - $int_part))`?

Comment (astj) [inline lib/Test/MockTime/HiRes.pm:82]: `sprintf('%.6f', $x)` rounds floating point number `$x`, and `int($x)` floors.

In a test written at https://github.com/plack/Plack/pull/551/files#diff-66e896132aa1d236dfae7cbf05485434 , after `sleep`ing 0.3 sec `tv_interval` becomes 0.2999..., not 0.3.  I guess this is due to perl's floating point number handling, and rounding by `sprintf` can avoid this problem.

``` perl
# Simplified version of https://github.com/plack/Plack/pull/551/files#diff-66e896132aa1d236dfae7cbf05485434

my $t1 = [Time::HiRes::gettimeofday()];
Time::HiRes::sleep 0.3;
my $t2 = [Time::HiRes::gettimeofday()];
warn scalar Time::HiRes::tv_interval($time, $now) * 1_000_000;
```

Comment (tarao) [inline lib/Test/MockTime/HiRes.pm:82]: I see, then, my real question was "why it is not `round(1_000_000 * ($Test::MockTime::fixed - $int_part))` where `round` is from [`Math::Round`](http://search.cpan.org/dist/Math-Round/Round.pm)?"  I'm worrying that the floating point number returned by `sprintf` may be converted internally during the operation of `*` to a native floating point number which cannot be expressed in 6 digits.  I guess this doesn't happen for a floating point number represented in a string but I'm not sure.  So please clarify this.

Comment (tarao) [inline lib/Test/MockTime/HiRes.pm:82]: Now I have another question.  My REPL for Perl 5.18.2 says that the value of `$result` in the following code is `1000000`.  Is this the intended behavior?

``` perl
my $t = 1.9999999999
my $result = 1_000_000 * sprintf("%.6f", $t - int($t))
```

Comment (tarao) [reply lib/Test/MockTime/HiRes.pm:82]: >  I guess this doesn't happen for a floating point number represented in a string but I'm not sure. So please clarify this.

Maybe we should bet that the guess is correct. Let's fix it when we lose the bet.

**Comment by astj** [comment]
Thanks!

**Pull request merged**